jsapp/js/api.ts

```diff
@@ -25,7 +33,54 @@ const fetchData = async <T>(path: string, method = 'GET', data?: Json) => {
     headers,
     body: JSON.stringify(data),
   });
+
   const contentType = response.headers.get('content-type');
+
+  if (!response.ok) {
+    // This will be returned with the promise rejection. It can include that
+    // response JSON, but not all endpoints/situations will produce one.
+    const failResponse: FailResponse = {
+      status: response.status,
+      statusText: response.statusText,
+    };
+
+    // For server issues, we simply reject with no additional data. We expect
+    // the UI to not react with notifications or other indicators.
+    if (
+      response.status === 401 ||
+      response.status === 403 ||
+      response.status === 404 ||
+      response.status >= 500
+    ) {
+      let errorMessage = t('An error occurred');
+      errorMessage += ' ';
+      errorMessage += response.status;
+      errorMessage += ' ';
+      errorMessage += response.statusText;
+
+      // We only want to display the server issues notification once, even with
+      // multiple calls failing
+      notifyServerErrorThrottled(errorMessage);
```

Comment:
I want to hold off on this for now—I thought about this some more and I don't throttling these is a good idea.

Two days ago, when we had a [double submit issue](https://github.com/kobotoolbox/kpi/pull/4538/commits/b7e6e3ee473bd27b5a27b8b72f406c289aae6d2d), the first indication was that there were 2 identical toast notifications.

Seeing lots of errors when one thing is broken isn't fun, but discarding errors without showing them at all is usually worse — particularly if the errors aren't duplicates and some may be more informative than others.

Related: Once toast notifications disappear, you can't review them at all. At least for our own debugging purposes, I want to start logging them to the browser console.